There is one unit test that fails, however it was failing before I changed anything. I had a look but couldn't figure out what was wrong:

Before my PR:
```
============================  Overall Tests Report  ============================
*** unit.modules.cron_test Tests  **********************************************
 --------  Failed Tests  -------------------------------------------------------
   -> unit.modules.cron_test.PsTestCase.test_set_special  ......................
       Traceback (most recent call last):
         File "/opt/local/lib/python2.7/site-packages/mock/mock.py", line 1305, in patched
           return func(*args, **keywargs)
         File "/salt_dev/salt/tests/unit/modules/cron_test.py", line 580, in test_set_special
           write_cron_lines_mock.assert_has_calls(expected_write_call)
         File "/opt/local/lib/python2.7/site-packages/mock/mock.py", line 969, in assert_has_calls
           ), cause)
         File "/opt/local/lib/python2.7/site-packages/six-1.9.0-py2.7.egg/six.py", line 692, in raise_from
           raise value
       AssertionError: Calls not found.
       Expected: call('DUMMY_USER', ['5 0 * * * /tmp/no_script.sh\n', '# Lines below here are managed by Salt, do not edit\n', '@hourly echo Hi!\n'])
       Actual: [call('DUMMY_USER', ['5 0 * * * /tmp/no_script.sh\n', '# Lines below here are managed by Salt, do not edit\n', '@hourly echo Hi!\n']),
        call().__getitem__('retcode'),
        call().__getitem__().__nonzero__(),
        call().__getitem__('stderr')]
   .............................................................................
 -------------------------------------------------------------------------------
================================================================================
FAILED (total=14, skipped=0, passed=13, failures=1, errors=0)
============================  Overall Tests Report  ============================
```

After my PR:
```
============================  Overall Tests Report  ============================
*** unit.modules.cron_test Tests  **********************************************
 --------  Failed Tests  -------------------------------------------------------
   -> unit.modules.cron_test.PsTestCase.test_set_special  ......................
       Traceback (most recent call last):
         File "/opt/local/lib/python2.7/site-packages/mock/mock.py", line 1305, in patched
           return func(*args, **keywargs)
         File "/salt_dev/salt/tests/unit/modules/cron_test.py", line 580, in test_set_special
           write_cron_lines_mock.assert_has_calls(expected_write_call)
         File "/opt/local/lib/python2.7/site-packages/mock/mock.py", line 969, in assert_has_calls
           ), cause)
         File "/opt/local/lib/python2.7/site-packages/six-1.9.0-py2.7.egg/six.py", line 692, in raise_from
           raise value
       AssertionError: Calls not found.
       Expected: call('DUMMY_USER', ['5 0 * * * /tmp/no_script.sh\n', '# Lines below here are managed by Salt, do not edit\n', '@hourly echo Hi!\n'])
       Actual: [call('DUMMY_USER', ['5 0 * * * /tmp/no_script.sh\n', '# Lines below here are managed by Salt, do not edit\n', '@hourly echo Hi!\n']),
        call().__getitem__('retcode'),
        call().__getitem__().__nonzero__(),
        call().__getitem__('stderr')]
   .............................................................................
 -------------------------------------------------------------------------------
================================================================================
FAILED (total=14, skipped=0, passed=13, failures=1, errors=0)
============================  Overall Tests Report  ============================
```

--- a/salt/modules/cron.py
+++ b/salt/modules/cron.py
@@ -155,7 +155,7 @@ def _get_cron_cmdstr(path, user=None):
     '''
     cmd = 'crontab'
 
-    if __grains__.get('os_family') not in ('Solaris', 'AIX') and user:
+    if user and __grains__.get('os_family') not in ('Solaris', 'AIX'):
         cmd += ' -u {0}'.format(user)
 
     return '{0} {1}'.format(cmd, path)

--- a/salt/modules/cron.py
+++ b/salt/modules/cron.py
@@ -221,7 +221,7 @@ def _write_cron_lines(user, lines):
     path = salt.utils.mkstemp()
     if __grains__.get('os_family') in ('Solaris', 'AIX') and appUser != user:
         # on solaris/aix we change to the user before executing the commands
-        with salt.utils.fpopen(path, 'w+', uid=__salt__['file.user_to_uid'](user)) as fp_:
+        with salt.utils.fpopen(path, 'w+', uid=__salt__['file.user_to_uid'](user), mode=600) as fp_:
             fp_.writelines(lines)
         ret = __salt__['cmd.run_all'](_get_cron_cmdstr(path, user),
                                       runas=user,

--- a/salt/modules/cron.py
+++ b/salt/modules/cron.py
@@ -221,13 +221,13 @@ def _write_cron_lines(user, lines):
     path = salt.utils.mkstemp()
     if __grains__.get('os_family') in ('Solaris', 'AIX') and appUser != user:
         # on solaris/aix we change to the user before executing the commands
-        with salt.utils.fpopen(path, 'w+', uid=__salt__['file.user_to_uid'](user), mode=600) as fp_:
+        with salt.utils.fpopen(path, 'w+', uid=__salt__['file.user_to_uid'](user), mode=0o600) as fp_:
             fp_.writelines(lines)
         ret = __salt__['cmd.run_all'](_get_cron_cmdstr(path, user),
                                       runas=user,
                                       python_shell=False)
     else:
-        with salt.utils.fopen(path, 'w+') as fp_:
+        with salt.utils.fpopen(path, 'w+', mode=0o600) as fp_:
             fp_.writelines(lines)
         ret = __salt__['cmd.run_all'](_get_cron_cmdstr(path, user),
                                       python_shell=False)

